File: mad_lib.exs
# Enter a noun: dog
# Enter a verb: walk
# Enter an adjective: blue
# Enter an adverb: quickly
# Do you walk your blue dog quickly? That's hilarious!

defmodule MadLib do
  def run do
    noun = get_input("Enter a noun: ")
    verb = get_input("Enter a verb: ")
    adjective = get_input("Enter an adjective: ")
    adverb = get_input("Enter an adverb: ")

    IO.puts("Do you #{verb} your #{adjective} #{noun} #{adverb}? That's hilarious!")
  end

  defp get_input(prompt) do
    IO.gets(prompt) |> String.trim()
  end
end

MadLib.run()

File: mad_lib.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func getInput(prompt string, reader *bufio.Reader) string {
	fmt.Print(prompt)
	input, _ := reader.ReadString('\n')
	return strings.TrimSpace(input)
}

func main() {
	reader := bufio.NewReader(os.Stdin)

	noun := getInput("Enter a noun: ", reader)
	verb := getInput("Enter a verb: ", reader)
	adjective := getInput("Enter an adjective: ", reader)
	adverb := getInput("Enter an adverb: ", reader)

	fmt.Printf("Do you %s your %s %s %s? That's hilarious!\n", verb, adjective, noun, adverb)
}
